docs/metrics.md

```diff
@@ -17,6 +20,99 @@ for many cases; some examples:
+
+### OpenTelemetry protocol
+
+To enable the OpenTelemetry protocol you must set the environment variable `OTEL_EXPORTER_OTLP_ENDPOINT` or `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`. It will not default to the default value if left blank, instead no Open
+
+You can configure the protocol using the environment variables documented in [standard environment variables](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/).
+
+The [configuration options](#common) in the controller ConfigMap `metricsTTL`, `options` and `temporality` affect the OpenTelemetry behavior, but the other parameters do not.
+
+To use the [OpenTelemetry collector](https://opentelemetry.io/docs/collector/) you can configure it
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+```
+
+You can use the [OpenTelemetry operator](https://opentelemetry.io/docs/kubernetes/operator/) to setup the collector and instrument the workflow-controller.
+
+### Prometheus scraping
+
+You can adjust various elements of the prometheus metrics configuration by changing values in the [Workflow Controller Config Map](workflow-controller-configmap.md).
+
+```yaml
+metricsConfig: |
+  # Enabled controls the prometheus metric. Default is true, set "enabled: false" to turn off
+  enabled: true
+
+  # Path is the path where prometheus metrics are emitted. Must start with a "/". Default is "/metrics"
+  path: /metrics
+
+  # Port is the port where prometheus metrics are emitted. Default is "9090"
+  port: 8080
+
+  # IgnoreErrors is a flag that instructs prometheus to ignore metric emission errors. Default is "false"
+  ignoreErrors: false
+
+  # Use a self-signed cert for TLS, default true
+  secure: true
+```
+
+The metric names emitted by this mechanism are prefixed with `argo_workflows_`. `Attributes` are exposed as prometheus `labels` of the same name.
+
+### Common
+
+You can adjust various elements of the metrics configuration by changing values in the [Workflow Controller Config Map](workflow-controller-configmap.md).
+
+```yaml
+metricsConfig: |
+  # MetricsTTL sets how often custom metrics are cleared from memory. Default is "0", metrics are never cleared
+  metricsTTL: "10m"
+  options:
+    pod_missing:
+      disable: true
+    cronworkflows_triggered_total:
+      disabledAttributes:
+        - name
+    k8s_request_duration:
+      histogramBuckets: [ 1.0, 2.0, 10.0 ]
+```
+
+### Options
+
+Using options you can manipulate the metrics created by the workflow controller. These options apply to the built-in metrics and any custom metrics you create. Each option applies to the named metric only, and applies to all output methods.
+
+`disable: true` will disable the emission of the metric from the system.
```

Comment:
Typically the convention in the ecosystem, such as Helm, is `enabled: false` -- note that this includes `metricsConfig.enabled` just above this.

The tense and convention is important to keep consistent and predictable IMO

Reply:
Yeah, this and the 'options' naming were just where I'd ended up and wanted to see if my changes passed in proper CI.